extension/src/NABfunctions.ts

```diff
@@ -1458,3 +1465,69 @@ function activeTextEditorIsXlf(): boolean {
     vscode.window.activeTextEditor.document.uri.fsPath.endsWith("xlf")
   );
 }
+
+export async function createPermissionSetForAllObjects(): Promise<void> {
+  logger.log("Running: createPermissionSetForAllObjects");
+  Telemetry.trackEvent("createPermissionSetForAllObjects");
+  try {
+    const appManifest = SettingsLoader.getAppManifest();
+    const settings = SettingsLoader.getSettings();
+    const allObjects = await WorkspaceFunctions.getAlObjectsFromCurrentWorkspace(
+      settings,
+      appManifest,
+      true,
+      false,
+      false
+    );
+    const firstNumber = appManifest.idRanges[0]
+      ? appManifest.idRanges[0].from
+      : 50000;
+    const prefix = getDefaultPrefix();
+    const permissionSet = new ALPermissionSet(
+      `${prefix}All`,
+      "All permissions",
+      firstNumber
+    );
+    allObjects
+      .filter(
+        (obj) =>
+          [
+            ALObjectType.codeunit,
+            ALObjectType.page,
+            ALObjectType.query,
+            ALObjectType.report,
+            ALObjectType.table,
+            ALObjectType.xmlPort,
+          ].includes(obj.objectType) && !obj.isObsolete()
+      )
+      .forEach((obj) => {
+        if (
+          obj.objectType === ALObjectType.table &&
+          obj.getProperty(ALPropertyType.tableType, ALTableType.normal) ===
+            ALTableType.normal
+        ) {
+          permissionSet.permissions.push(
+            new ALPermission(ALObjectType.tableData, obj.objectName, "RIMD")
+          );
+        }
+        permissionSet.permissions.push(
+          new ALPermission(obj.objectType, obj.objectName, "X")
+        );
+      });
+    const filePath = path.join(
+      settings.sourceFolderPath,
+      `All.PermissionSet.al`
+    );
+    if (fs.existsSync(filePath)) {
+      throw new Error(`File ${filePath} already exists.`);
+    }
```

Comment:
Maybe do this check before actually creating all the permissions?

Reply:
fixed